arm-web/2016-08-01/swagger/WebApps.json

```diff
@@ -13219,195 +13379,450 @@
-          "type": "string"
-        },
-        "unauthenticatedClientAction": {
-          "description": "The action to take when an unauthenticated client attempts to access the app.",
+        "properties": {
+          "description": "SiteAuthSettings resource specific properties",
+          "properties": {
+            "enabled": {
+              "description": "<code>true</code> if the Authentication / Authorization feature is enabled for the current app; otherwise, <code>false</code>.",
+              "type": "boolean"
+            },
+            "runtimeVersion": {
+              "description": "The RuntimeVersion of the Authentication / Authorization feature in use for the current app.\n The setting in this value can control the behavior of certain features in the Authentication / Authorization module.",
+              "type": "string"
+            },
+            "unauthenticatedClientAction": {
+              "description": "The action to take when an unauthenticated client attempts to access the app.",
+              "enum": [
+                "RedirectToLoginPage",
+                "AllowAnonymous"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "UnauthenticatedClientAction",
+                "modelAsString": false
+              }
+            },
+            "tokenStoreEnabled": {
+              "description": "<code>true</code> to durably store platform-specific security tokens that are obtained during login flows; otherwise, <code>false</code>.\n The default is <code>false</code>.",
+              "type": "boolean"
+            },
+            "allowedExternalRedirectUrls": {
+              "description": "External URLs that can be redirected to as part of logging in or logging out of the app. Note that the query string part of the URL is ignored.\n This is an advanced setting typically only needed by Windows Store application backends.\n Note that URLs within the current domain are always implicitly allowed.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "defaultProvider": {
+              "description": "The default authentication provider to use when multiple providers are configured.\n This setting is only needed if multiple providers are configured and the unauthenticated client\n action is set to \"RedirectToLoginPage\".",
+              "enum": [
+                "AzureActiveDirectory",
+                "Facebook",
+                "Google",
+                "MicrosoftAccount",
+                "Twitter"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "BuiltInAuthenticationProvider",
+                "modelAsString": false
```

Comment:
Setting modelAsString to false can result in breaking changes whenever you modify this enum